src/Razor/src/Microsoft.NET.Sdk.Razor/ApplicationPartsProvider.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+
+namespace Microsoft.AspNetCore.Razor.Tasks
+{
+    public class ApplicationPartsProvider
```

Comment:
Making this `internal` is a bit tricky. I want to add unit tests for this, but the test project already has IVT access to some common types through Razor.Tools. I don't think we expect users to reference this assembly in any way, so leaving it public isn't the end of the world

Reply:
If you're going to leave it public can you make it have a non-shameful name 😆

Reply:
`MvcReferenceResolver`? This thing used to be called `ApplicationAssembliesProvider`. I thought I'd already made an improvement when I was moving it over.

Reply:
`ApplicationPartsProvider` is probably fine.